rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -3391,6 +3392,46 @@ private ABICallSpec getAddMultiKeysVote(FederationMember federationMember) {
         }
     }
 
+    @Test
+    void getProposedFederation_whenStorageProviderReturnsEmpty_shouldReturnEmpty() {
+        // arrange
+        activations = ActivationConfigsForTest.all().forBlock(0L);
+
+        storageProvider = mock(FederationStorageProvider.class);
+        when(storageProvider.getProposedFederation(federationMainnetConstants, activations)).thenReturn(Optional.empty());
+
+        federationSupport = federationSupportBuilder
+            .withFederationConstants(federationMainnetConstants)
+            .withFederationStorageProvider(storageProvider)
+            .withActivations(activations)
+            .build();
+
+        // act
+        Optional<Federation> actualProposedFederation = federationSupport.getProposedFederation();
+
+        // assert
+        assertFalse(actualProposedFederation.isPresent());
+    }
+
+    @Test
+    void getProposedFederation_whenStorageProviderReturnsProposedFederation_shouldReturnProposedFederation() {
+        // arrange
+        activations = ActivationConfigsForTest.all().forBlock(0L);
+        Federation proposedFederation = new P2shErpFederationBuilder().build();
+
+        storageProvider = mock(FederationStorageProvider.class);
```

Comment:
Let's create the actual storage object, using InMemoryStorage. The less mocks the better

Reply:
done